internal/hardware/validator_test.go

```diff
@@ -857,6 +858,35 @@ var _ = Describe("Disk eligibility", func() {
 		Expect(secondCallReasons).To(HaveLen(1))
 		Expect(secondCallReasons[0]).To(Equal(expectedMsg))
 	})
+
+	It("Check precise error messages for operator disk additions", func() {
+		cluster.OpenshiftVersion = "4.14"
+
+		// Mock operator that adds 7 GB to base requirement
+		operatorRequirement := &models.OperatorHostRequirements{
+			OperatorName: "test-operator",
+			Requirements: &models.ClusterHostRequirementsDetails{
+				DiskSizeGb: 7, // This creates 107 GB total (100 + 7)
+			},
+		}
+
+		operatorsMock.EXPECT().GetRequirementsBreakdownForHostInCluster(gomock.Any(), gomock.Any(), gomock.Any()).
+			Return([]*models.OperatorHostRequirements{operatorRequirement}, nil).AnyTimes()
+
+		By("Testing precise humanization with overhead scenario")
+		testDisk.SizeBytes = int64(99900000000) // 99.9 GB with overhead
+		testDisk.InstallationEligibility.NotEligibleReasons = []string{}
+
+		reasons, err := hwvalidator.DiskIsEligible(ctx, &testDisk, infraEnv, &cluster, &host, inventory)
+
+		Expect(err).ToNot(HaveOccurred())
+		Expect(reasons).To(HaveLen(1))
+		// The key improvement: precise function shows "99.9 GB" instead of "100 GB"
+		// This makes it clear the disk is slightly less than 100 GB (not exactly 100)
+		Expect(reasons[0]).To(ContainSubstring("99.9 GB"))
+		// Note: Operator addition only applies to cluster validation, not infraEnv validation
+		// So for this test, we expect the base 100 GB requirement
```

Comment:
not sure if the comments are useful in this context, I need to be convinced in the PR description, not in the comments of a code that will be merged.